src/features/auth/authSlice.js

```diff
@@ -0,0 +1,60 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import authenticateUser from "../../utils/authUtils";
+
+const INITIAL_STATE = {
+  loading: false,
+  authUser: JSON.parse(localStorage.getItem("authUser")),
+  error: "",
+  success: false,
+};
+
+export const loggedInUser = createAsyncThunk(
+  "auth/loggedInUser",
+  async (user) => {
+    try {
+      const authUser = await authenticateUser(user);
+
+      if (authUser) {
+        localStorage.setItem("authUser", JSON.stringify(authUser));
+        return authUser;
+      } else {
+        throw new Error("Invalid username or password");
+      }
+    } catch (error) {
+      throw error.message;
+    }
+  }
```

Comment:
Not providing user or password is not a app error. Show appropriate message to user when the user is not found. Try and catch is used to catch error that may happen during the ajax call or if BE returns an error


```suggestion
  async (user) => {
  let authUser = null
  try {
    authUser = await authenticateUser(user);
  } catch(e) {
    throw error.message;
  }
  
 if (authUser) {
      localStorage.setItem("authUser", JSON.stringify(authUser));
      return authUser;
 }

  showError("Invalid username or password");
  }
```